app/src/main/java/com/denisatrif/truthdare/viewmodel/PlayersViewModel.kt

```diff
@@ -43,19 +41,35 @@ class PlayersViewModel @Inject constructor(private val playersRepository: Player
         return liveData
     }
 
-    fun startGame() {
-        addAllPlayers()
-    }
-
-    private fun deleteAll() {
+    fun deletePlayer(player: Player) {
+        _playersList.remove(player)
         viewModelScope.launch(Dispatchers.IO) {
-            playersRepository.deleteAll()
+            playersRepository.delete(player)
         }
     }
 
-    fun deletePlayer(player: Player) {
+    fun getAllIds(): Flow<List<Int>> {
+        return playersRepository.getListOfIds()
+    }
+
+    fun getNext(id: Int): Flow<Player> {
+        return playersRepository.getPlayerWithId(id)
+    }
+
+    private fun getCount(): LiveData<Int> {
+        val liveData = MutableLiveData<Int>()
         viewModelScope.launch(Dispatchers.IO) {
-            playersRepository.delete(player)
+            val count = playersRepository.getCount()
+            liveData.postValue(count)
         }
+        return liveData
+    }
+
+    fun saveNumberOfPlayers() {
+        val count = getCount().value
+        val prefs = app.getSharedPreferences(
+            app.getString(R.string.preference_file_key), Application.MODE_PRIVATE
+        )
+        prefs?.edit()?.putInt("playersCount", count ?: 0)?.apply()
     }
```

Comment:
By the way, why do we even write the players' count in the SharedPreferences? I don't se this value read anywhere in the project.
And even if we do need it, can't we just always read it from the DB? And get rid of SharedPrefs altogether?